Fix self-energy for the superconductive case on the real axis

lcrippa opened the pull request:
The calculation of the self-energy on the real axis for bath topologies different from normal has some problems, which come from the way the Nambu basis matrix is constructed from G and F.

In particular, the issue comes from the lower left block of the Nambu basis matrix (Fbar). From direct calculation via Lehmann representation, it turns out that the relation between Fbar and F is the following (a,b orbitals):

Nambu block (1,2): F_{a, b}(\omega + i\epsilon)
Nambu block (2,1): F_{b, a}(\omega - i\epsilon)^*


This can of course have some simplifications in the case of singlet pairing, however we ran into problems in the following cases:

- hybrid and replica/general bath: the (2,1) block was not transposed
- replica/general bath: now, the Delta function can be complex. Therefore it is not in general true anymore that F(w-ie)* =F(w+ie)


This branch has the following fix:

- get_impF_superc has an optional argument zconj. If this is set to .true., zeta is transformed to zeta*
- F12 and F21 are calculated independently in get_Sigma_superc and get_Self_superc
- The (2,1) Nambu block is constructed always (for bath != normal) as transpose(conjg(F21(ispin,ispin,:,:,i)))

This is the most general case and it works always. Simplifications coming from the singlet pairing may be possible, but we have to be careful with the bath geometry. the anomalous bath H can be complex with replica/general bath.

lcrippa committed:
Correct superconductive Sigma on the real axis

lcrippa committed:
Fix imsign

lcrippa committed:
Better print for superc observables

lcrippa committed:
Simplified impF calculation logic

lcrippa committed:
Fix sign in fdelta in ED_FIT, to be consistent with https://github.com/EDIpack/EDIpack/commit/cfbe4b7965ffe419d886cac34710b17ac7818c24

lcrippa committed:
Updated comments and docs

lcrippa committed:
Remove print_logo function from documentation

aamaricci reviewed (approved):
Awesome! 
You guys did a great job.

lcrippa committed:
5.3.0 Update

Point version release. Changelog w.r.t 5.2.0:

- Enabled 0 bath sites
- Fixed real-axis superconductive Sigma
- Updated docs
- Print logo
- Minor improvements and bugfixes
